packages/runtimes/js/src/JSRuntime.js

```diff
@@ -347,6 +347,10 @@ function getLoaderRuntime({
   }
 
   let externalBundles = bundleGraph.getBundlesInBundleGroup(bundleGroup);
+  externalBundles = externalBundles.filter(
+    externalBundle => externalBundle.target == bundle.target,
+  );
```

Comment:
One question is how bundles from multiple different targets are getting into the same bundle groups in the first place?

Reply:
I believe I fixed it properly now 😅 @devongovett , added an updated graph pic as well where theres two separate bundlegroups for each target :)